pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/Producer.java

```diff
@@ -122,42 +124,28 @@ public void publishMessage(long producerId, long sequenceId, ByteBuf headersAndP
     }
 
     private boolean verifyChecksum(ByteBuf headersAndPayload) {
-        MessageMetadata metadata = null;
-        int readerIndex = headersAndPayload.readerIndex();
-        try {
-            metadata = Commands.parseMessageMetadata(headersAndPayload);
-
-            if (!metadata.hasChecksum()) {
-                // if we do not have the checksum, we do not verify checksum and return true
-                if (log.isDebugEnabled()) {
-                    log.debug("[{}] [{}] Payload does not have checksum to verify", topic, producerName);
-                }
-                return true;
-            }
-
-            if (metadata.hasCompression()) {
-                // if the message is compressed, we do not verify checksum and return true
-                if (log.isDebugEnabled()) {
-                    log.debug("[{}] [{}] Payload is compressed, not verifying checksum", topic, producerName);
+        
+        if (hasChecksum(headersAndPayload)) {
+            int checksum = readChecksum(headersAndPayload).intValue();
+            int readerIndex = headersAndPayload.readerIndex();
+            try {
+                long computedChecksum = computeChecksum(headersAndPayload);
+                if (checksum == computedChecksum) {
+                    return true;
+                } else {
+                    log.error("[{}] [{}] Failed to verify checksum", topic, producerName);
```

Comment:
actually, at this point we haven't persisted the message so, we don't have message-id so, we are not logging message-id.